Marlin/src/gcode/lcd/M73.cpp

```diff
@@ -39,7 +39,11 @@
  */
 void GcodeSuite::M73() {
   if (parser.seen('P'))
-    ui.set_progress(parser.value_byte());
+    #if HAS_PRINT_PROGRESS_PERMYRIAD
+      ui.set_progress((uint16_t)(parser.value_float() * 100));
+    #else
+      ui.set_progress(parser.value_byte());
+    #endif
```

Comment:
G-codes can't change the meaning of their parameters without also exposing this fact to hosts and users. Thus you will have to use a parameter other than 'P' for this new feature.